src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1890,3 +1892,24 @@ fn is_primitive(path_str: &str, ns: Namespace) -> Option<(&'static str, Res)> {
 fn primitive_impl(cx: &DocContext<'_>, path_str: &str) -> Option<&'static SmallVec<[DefId; 4]>> {
     Some(PrimitiveType::from_symbol(Symbol::intern(path_str))?.impls(cx.tcx))
 }
+
+fn strip_generic_parameters(path_str: &str) -> String {
+    let mut stripped_path_string = String::new();
+
+    let mut param_depth = 0;
+
+    for c in path_str.chars() {
+        if c == '<' {
+            param_depth += 1;
+        } else if c == '>' {
+            param_depth -= 1;
+        } else if param_depth == 0 {
+            stripped_path_string.push(c)
+        }
+    }
+
+    // Should we throw an error here if `param_depth` != 0? That would mean the link is malformed,
+    // e.g. `Vec<T` or `Vec<T>>` or something like that.
+
+    stripped_path_string
+}
```

Comment:
Alternatively, this could be implemented much more simply by just skipping everything after the first `<`. One advantage of this parser-esque implementation is that we can be smarter about the links and report errors if the link is malformed.

Another advantage is that parts of it could probably be extracted in the future for an implementation like the one described in https://github.com/rust-lang/rust/issues/62834#issuecomment-513705453.

Reply:
How would that handle things like `Vec::<T>::new`? I think the current implementation is wrong for that too, it will return `Vec::::new` which isn't correct.

A better possible approach is to split `path_str` into segments (`path_str.split("::")`) and remove the generics from each segment individually. If the segment ends up as empty, don't add it to the string.

Reply:
Implemented in e367321 🎉 !